indico/htdocs/js/indico/jquery/itempicker.js

```diff
@@ -0,0 +1,162 @@
+
+        options: {
+            onSelect: null, // function called every time an item gets selected from the list
+            createItem: null, // function returning a jQuery object representing each item in the dropdown
+            items: null, // all the options that should be included in the dropdown
+            createCustomFooterElements: null,
+            containerClasses: '', // extra classes to be applied to the container
+            filterPlaceholder: '' // text displayed inside the input field
+        },
+
+        _create: function() {
+            var items = this.options.items || this.element.data('items') || [];
+            var qtipContent = this._createQtipContent(items);
+
+            this.element.qtip({
+                prerender: false,
+                overwrite: false,
+                style: {
+                    classes: 'item-picker-qtip ' + this.options.containerClasses
+                },
+                position: {
+                    my: 'right top',
+                    at: 'left top',
+                    target: this.element,
+                    adjust: {
+                        mouse: false,
+                        scroll: false
+                    }
+                },
+                content: {
+                    text: qtipContent
+                },
+                show: {
+                    event: 'click',
+                    solo: true
+                },
+                hide: {
+                    event: 'unfocus click'
+                },
+                events: {
+                    show: function() {
+                        _.delay(function() {
+                            qtipContent.find('.filter-input').focus();
+                        });
+                    }
+                }
+            });
+        },
+        _destroy: function() {
+            self.element.qtip('destroy');
+        },
+        _createQtipContent: function(items) {
+            var self = this;
+            var filterPlaceholder = self.options.filterPlaceholder || $T.gettext('Search');
+            var dropdownContainer = $('<div>', {'class': 'dropdown-container'});
+            var filterWrapper = $('<div>', {'class': 'dropdown-filter-wrapper'});
+            var filterInput = $('<input>', {'type': 'text', 'class': 'filter-input',
+                                            'css': {'height': 30, 'line-height': 30},
+                                            'attr': {'placeholder': filterPlaceholder}});
+            var itemsContainer = $('<div>', {'class': 'dropdown-items-container'});
+
+            function handleInput() {
+                var textTyped = filterInput.val().trim().toLowerCase();
+
+                dropdownContainer.find('.dropdown-item').each(function() {
+                    var $item = $(this);
+
+                    $item.closest('.dropdown-wrapper-item')
+                         .toggle($item.data('filter').toLowerCase().indexOf(textTyped) !== -1);
```

Comment:
I think it'd be more readable if you put the toggle in the previous line and used a separate variable for the state

    var found = !!~$item.data('filter').toLowerCase().indexOf(textTyped);

(or keep `!== -1` if you prefer that)